### PR TITLE
Add unit tests for CaptureBuffer

### PR DESCRIPTION
**Pull request by AlekseyMartynov**
## Summary by CodeRabbit

* **New Features**
  * Added configurable audio capture formats and maximum durations.
  * Audio processing now adapts to the selected format’s sample rate.
  * The default configuration remains 16-bit mono audio with a 12-second duration.

* **Bug Fixes**
  * Improved handling of short streams, including complete consumption and zero-padding.
  * Preserved reliable stopping, disposal, truncation, and maximum-duration behavior.

* **Tests**
  * Expanded coverage for asynchronous stream consumption and capture-buffer edge cases.

### CODE REVIEW

File: CaptureHelpers/CaptureBuffer.cs [hunk 1]
```diff
@@ -30,16 +27,24 @@ int RemainingCount {
...
-        WaveBuffer = new(CaptureHelper.WAVE_FORMAT, MaxDuration) {
+    public CaptureBuffer()
+        : this(CaptureHelper.WAVE_FORMAT, TimeSpan.FromSeconds(12)) {
+        // 12 sec is max 'retryMs' returned by Shazam API
+    }
+
+    public CaptureBuffer(WaveFormat waveFormat, TimeSpan maxDuration) {
+        Debug.Assert(waveFormat.Encoding == WaveFormatEncoding.Pcm);
+        Debug.Assert(waveFormat.BitsPerSample == 16);
+        Debug.Assert(waveFormat.Channels == 1);
```
[the remark this answers is not in the document]
Comment (AlekseyMartynov) [reply CaptureHelpers/CaptureBuffer.cs:38]: @coderabbitai
Thanks. Since this is an app, not a library, and production always uses CaptureHelper.WAVE_FORMAT, I'd keep these checks debug-only. Sounds goods?

Looking through the code, I think I'll add another test case for a short stream to cover 'break; // end of stream'

File: Test/CaptureBufferTests.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+using Xunit;
+
+namespace Project.Test;
+
+public class CaptureBufferTests {
+    const int TestSampleRate = 123;
+    const short PositiveHalfShort = short.MaxValue / 2;
+    const float ReadSentinel = 0xdeadbeef;
+
+[Fact]
```
[the remark this answers is not in the document]
Comment (AlekseyMartynov) [reply Test/CaptureBufferTests.cs:16]: Yes, pushed 2d85af7df62c4a00dedf31379867b246a4e674cf
